sahi/predict.py

```diff
@@ -460,14 +472,97 @@ def predict(
     # iterate over source images
     durations_in_seconds["prediction"] = 0
     durations_in_seconds["slice"] = 0
-    for ind, image_path in enumerate(tqdm(image_path_list, "Performing inference on images")):
+
+    # Input video
+    if export_video:
+        if video_path is not None:
+            #get video name with extension
+            image_base = os.path.basename(video_path)
+        else:
+            print("\nVideo path is none. Enter (--video_path 'path' --export_video) in terminal!\n")
+        # get video from video path
+        cap = cv2.VideoCapture(video_path)
+        # get video properties
+        fps = cap.get(cv2.CAP_PROP_FPS)
+        w = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+        h = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        size = (w, h)
+
+        fourcc = cv2.VideoWriter_fourcc(*'mp4v')
+        out = cv2.VideoWriter(str(save_dir) + '/{}'.format(image_base), fourcc, fps, size)
```

Comment:
`save_dir` is already `pathlib.Path` object, you can append the path like, and moreover this type of manual joining as strings can be problematic.

```python
new_path = save_dir / "my_new_path"
```

Reply:
Do you want to I use path like: `new_path = save_dir / "my_new_path"` `out = cv2.VideoWriter(new_path, fourcc, fps, size) ` or `new_path = os.path.join(save_dir, image_base`